js/alerts.js

```diff
@@ -53,41 +100,106 @@ function loadSelectors() {
 
     });
 
-    $('#button').click(function () {
-        var rev = $('#rev').val();
-        var test = $('#test').val();
-        var platform = $('#platform').val();
-        var showall = 0;
-        var testIndex = $("select[id='test'] option:selected").index();
-        var platIndex = $("select[id='platform'] option:selected").index();
-        if ($('#checkbox').is(":checked")) {
-            console.log("checked");
-            showall = 1;
-        }
-        document.cookie = "platform = " + platform;
-        document.cookie = "test = " + test;
+    $('#button').bind("click", filterOnClick);
+}
 
-        var href = "alerts.html";
-        var flag = '?';
-        if (rev && rev != '') {
-            href += flag + "rev=" + rev;
-            flag = '&';
+//used for lexical sorting
+function compare(a, b) {
+    a = a.toString().toLowerCase();
+    b = b.toString().toLowerCase();
+    if (a < b)
+        return -1;
+    if (a > b)
+        return 1;
+    return 0;
+}
+
+function resetOptions(data,id) {
+    for (var i in data) {
+        var newoption = document.createElement("option");
+        newoption.id = id;
+        var value = data[i];
+        $("#"+id).append("<option value=\"" + value + "\">" + value + "</option>");
+    }
+}
+
+function updateSelectors(changedElementId, callback) {
+    
+    //to prevent selecting the button while the options are being updated [ to avoid irregular indexes of options ]
+    $('#button').unbind("click", filterOnClick).attr('disabled','disabled');
+    
+    rev = $('#rev').val() || "";
+    test = $('#test').val() || "";
+    platform = $('#platform').val() || "";
+    
+    if ( callback && typeof(callback)==="function") {
```

Comment:
spacing issues here, should be:
if (callback && typeof(callback) === "function") {